From: Enze Li <lienze@kylinos.cn>
Date: Fri, 15 Aug 2025 17:21:10 +0800
Subject: [PATCH] mm/damon/Kconfig: make DAMON_STAT_ENABLED_DEFAULT depend on
 DAMON_STAT

The DAMON_STAT_ENABLED_DEFAULT option is strongly tied to DAMON_STAT
option -- enabling it alone is meaningless.  This patch makes
DAMON_STAT_ENABLED_DEFAULT depend on DAMON_STAT, ensuring functional
consistency.

Signed-off-by: Enze Li <lienze@kylinos.cn>
Link: https://patch.msgid.link/20250815092110.811757-1-lienze@kylinos.cn
Fixes: 369c415e6073 ("mm/damon: introduce DAMON_STAT module")
Reviewed-by: SeongJae Park <sj@kernel.org>
Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/damon/Kconfig b/mm/damon/Kconfig
index b3171f9406c1..8c868f7035fc 100644
--- a/mm/damon/Kconfig
+++ b/mm/damon/Kconfig
@@ -104,7 +104,7 @@ config DAMON_STAT
 
 config DAMON_STAT_ENABLED_DEFAULT
 	bool "Enable DAMON_STAT by default"
-	depends on DAMON_PADDR
+	depends on DAMON_STAT
 	default DAMON_STAT
 	help
 	  Whether to enable DAMON_STAT by default.  Users can disable it in
-- 
2.39.5

